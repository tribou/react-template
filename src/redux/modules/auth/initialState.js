// @flow
import type { AuthState } from './types'

// Initial state with default values
const initialState: AuthState = {
  token: '',
  error: '',
  isFetching: false,
}

export default initialState
