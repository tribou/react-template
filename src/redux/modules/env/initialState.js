// @flow
const initialState: Object = {}

export default initialState
