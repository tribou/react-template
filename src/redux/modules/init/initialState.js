// @flow
const initialState = {
  isLoading: true,
  loaded: false,
}

export default initialState
