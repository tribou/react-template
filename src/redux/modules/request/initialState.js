// @flow
const initialState = { userAgent: '' }

export default initialState
