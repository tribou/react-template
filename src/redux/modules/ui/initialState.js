// @flow
import type { UIState } from './types'

const initialState: UIState = {
  sidebarOpen: true,
  error: null,
  // Manual option to show loading indicator if needed
  showLoading: false,
}

export default initialState
