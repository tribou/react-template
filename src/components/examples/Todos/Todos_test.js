// @flow

// Enzyme docs:
// http://airbnb.io/enzyme/docs/api/index.html

import React from 'react'
import { shallow } from 'enzyme'
import initialState from 'src/redux/modules/examples/todos/initialState'
import Todos from './Todos'

const mockProps = {
  setFilterCurrent: jest.fn(),
  setFilterDone: jest.fn(),
  getTodos: jest.fn(),
}

it('<Todos> renders the Todos content', () => {

  const todos = initialState.list
  const filter = initialState.filter

  const wrapper = shallow(
    <Todos
      {...mockProps}
      todos={todos}
      filter={filter}
    />
  )

  expect(wrapper).toMatchSnapshot()

})
