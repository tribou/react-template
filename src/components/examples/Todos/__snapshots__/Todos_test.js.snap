// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todos> renders the Todos content 1`] = `
<div
  className="todos"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Todos"
  />
  <div
    className="content"
  >
    Todos page
    <ul />
    <Link
      replace={false}
      to="/home"
    >
      Back
    </Link>
    <button
      onClick={[Function]}
    >
      Filter Done
    </button>
    <button
      onClick={[Function]}
    >
      Filter Current
    </button>
  </div>
</div>
`;
